hasql/asyncsqlalchemy.py

```diff
@@ -35,8 +35,9 @@ async def _is_master(self, connection: AsyncConnection):
         return result
 
     async def _pool_factory(self, dsn: Dsn):
-        # TODO: Add support of psycopg3 after release of sqlalchemy 2.0
-        d = str(dsn).replace("postgresql", "postgresql+asyncpg")
+        d = str(dsn)
+        if d.startswith('postgresql://'):
+            d = d.replace('postgresql://', 'postgresql+asyncpg://', 1)
```

Comment:
This line is needed to maintain backward compatibility, but at the same time allow the use of psycopg3